dev/tasks/conda-recipes/arrow-cpp/build-pyarrow.sh

```diff
@@ -16,10 +16,26 @@ export PYARROW_WITH_ORC=1
 export PYARROW_WITH_PARQUET=1
 export PYARROW_WITH_PLASMA=1
 export PYARROW_WITH_S3=1
+BUILD_EXT_FLAGS=""
+
+# Enable CUDA support
+if [[ ! -z "${cuda_compiler_version+x}" && "${cuda_compiler_version}" != "None" ]]
+then
+    export PYARROW_WITH_CUDA=1
+    BUILD_EXT_FLAGS="${BUILD_EXT_FLAGS} --with-cuda"
```

Comment:
Passing env var shouldn't be sufficient?

Reply:
Yes, it is. Pushed the change here and [upstream](https://github.com/conda-forge/arrow-cpp-feedstock/pull/156)